### PR TITLE
feat: add v0.functions

### PR DESCRIPTION
**Pull request by Endebert**

### CODE REVIEW

File: src/v0/functions.ts [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+          uri = queryOrOptions.uri
+        } else {
+          let queryString = ''
+          if (queryOrOptions && (queryOrOptions.query || queryOrOptions.limit)) {
+            queryString = qs.stringify({ limit: queryOrOptions.limit, ...queryOrOptions.query })
+          }
+
+          uri = `${this.options.base}${this.endpoint}/${this.options.user}${
+            queryString ? `?${queryString}` : ''
+            }`
```
Comment (erezsob) [inline src/v0/functions.ts]: Is there a specific reason you're not using the `UriHelper` here?

Comment (Endebert) [reply src/v0/functions.ts]: I copy-pasted the devices endpoint and replaced device with function. Should I change it to use UriHelper?

File: src/v0/functions.ts [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+  constructor(public message: string = 'Could not delete function', properties?: any) {
+    super(message, properties)
+  }
+}
+export class FunctionBindingFailed extends BaseError {
+  public name = 'FunctionBindingFailed'
+  constructor(public message: string = 'Could not bind function', properties?: any) {
+    super(message, properties)
+  }
+}
```
Comment (erezsob) [inline src/v0/functions.ts]: Some of the error classes don't seem to be used. Is that also intentional?

Comment (Endebert) [reply src/v0/functions.ts]: ah good catch, yeah I'll remove the unused ones.

File: src/v0/functions.ts (hunk 1 above)
Comment (erezsob) [reply src/v0/functions.ts]: Yes, please replace. It's legacy and we don't want to do stuff this way anymore.

**Comment by erezsob** [comment]
Also, not tests? 😢

**Comment by Endebert** [comment]
added tests for getting functions

**Pull request merged**